pkg/k8shandler/collection.go

```diff
@@ -50,6 +54,23 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCollection() (err err
 			return
 		}
 
+		if collectorConfig, err = clusterRequest.generateCollectorConfig(); err != nil {
+			return
+		}
+		logger.Debugf("Generated collector config: %s", collectorConfig)
+		collectorConfHash = utils.CalculateMD5Hash(collectorConfig)
+	} else {
+		if err = clusterRequest.RemoveServiceAccount("logcollector"); err != nil {
+			return
+		}
+
+		if err = clusterRequest.RemovePriorityClass(clusterLoggingPriorityClassName); err != nil {
+			return
+		}
+	}
+
+	if cluster.Spec.Collection.Logs.Type == logging.LogCollectionTypeFluentd {
```

Comment:
why is this broken out from the above if block?

Reply:
hold over from rsyslog impl.  I'll move it